### ISSUE TITLE
[FEA]: Test cudf 22.12

### ISSUE THREAD

**Issue by dagardner-nv** [nv-morpheus/Morpheus#550]
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

Critical (currently preventing usage)

### Please provide a clear description of problem this feature solves

Test cudf 22.12 with Morpheus.

Things to test with this version:
1. Is it faster?
2. Does it work with the experimental engine?
4. Can we remove our work-around for handling slashes

### Describe your ideal solution

n/a

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open feature requests](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22%2Cimprovement%2Cenhancement) and have found no duplicates for this feature request

**Comment by dagardner-nv** [comment]
Unfortunately I wasn't able to build Morpheus with cudf-22.12, I ran into too many version conflicts and quickly discovered that testing against an updated cudf wasn't going to be a simple task.

What I did instead was create a test to perform many cudf tasks commonly used by Morpheus they are:
- Load a CSV file
- Load a JSON file
- Filter null values
- Insert a new column
- Perform a slice operation.

What I found was there was no meaningful performance difference between 22.08 and 22.12. However 22.12 was slightly slower, but the differences were negledgable.

My CPU/GPU info:
```
$ lscpu 
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   43 bits physical, 48 bits virtual
CPU(s):                          64
On-line CPU(s) list:             0-63
Thread(s) per core:              2
Core(s) per socket:              32
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       AuthenticAMD
CPU family:                      23
Model:                           49
Model name:                      AMD Ryzen Threadripper PRO 3975WX 32-Cores
Stepping:                        0
Frequency boost:                 enabled
CPU MHz:                         3500.000
CPU max MHz:                     4368.1641
CPU min MHz:                     2200.0000
BogoMIPS:                        6986.49
Virtualization:                  AMD-V
L1d cache:                       1 MiB
L1i cache:                       1 MiB
L2 cache:                        16 MiB
L3 cache:                        128 MiB
NUMA node0 CPU(s):               0-63
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Mitigation; untrained return thunk; SMT enabled with STIBP protection
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, STIBP always-on, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt p
                                 dpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 ss
                                 e4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch 
                                 osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibpb stibp 
                                 vmmcall fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_
                                 occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_cle
                                 an flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip rdpid overflow_recov succor smca 
                                 sev sev_es
$ nvidia-smi 
Fri Jan  6 10:42:13 2023       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 520.61.05    Driver Version: 520.61.05    CUDA Version: 11.8     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA RTX A6000    On   | 00000000:61:00.0  On |                  Off |
| 30%   32C    P8    33W / 300W |   1284MiB / 49140MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1613      G   /usr/lib/xorg/Xorg                740MiB |
|    0   N/A  N/A      2012      G   xfwm4                               6MiB |
|    0   N/A  N/A    912136      G   ...013901808850016359,131072      314MiB |
|    0   N/A  N/A    912294      G   ...veSuggestionsOnlyOnDemand       83MiB |
|    0   N/A  N/A    914595      G   ...RendererForSitePerProcess      137MiB |
+-----------------------------------------------------------------------------+
```

**Comment by dagardner-nv** [comment]
Re-opening, I forgot to test serialization, and the experimental engine.

**Comment by dagardner-nv** [comment]
I re-ran the test, testing for write speeds. What I found was that in writing a large df (331MB of CSV data) to JSON had a 2x improvement, while writing another large df (161MB of JSON data) to JSON ended up being much slower in 22.12.

I tested setting the `engine` argument, however since I was already setting `lines=True` in 22.08 this always results in `engine=cudf` and the same is true in 22.12.
